### PR TITLE
DrivAerML: 4L/384d + lr=4e-4 — fill LR gap at 384d

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The optimal learning rate for 4L/384d may differ from the 4L/256d baseline. The current best uses lr=5e-4 (PR #2602). edward PR #2693 tests lr=3e-4. This experiment fills the gap at lr=4e-4 to map the LR response curve for the wider 384d model. The extra width increases the effective learning rate needed — or the reverse — and lr=4e-4 sits precisely in the unexplored middle of this range.

## Instructions

Run the 4L/384d baseline configuration with `--lr 4e-4` instead of the default 5e-4. No other changes.

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 4e-4 --cosine-t-max 30 \
  --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 384 --model-heads 6 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --agent zoro --wandb-name "zoro/drivaerml-4L384d-lr4e4" \
  --wandb-group "drivaerml-4L384d-variants"
```

Report the best `val_primary/surface_rel_l2_pct` and the epoch it was reached.

## Baseline

- **DrivAerML best:** 5.73% (val) at epoch 144 — PR #2602 (4L/384d, T_max=30, lr=5e-4)
- **In-flight:** edward PR #2693 testing lr=3e-4 at 4L/384d
- **External target:** <3.71% (AB-UPT, ~500 epochs)
- **W&B group:** drivaerml-4L384d-variants (for easy comparison with lr=3e-4 and lr=5e-4 runs)
- **Reproduce (baseline):**
  ```bash
  cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 384 --model-heads 6 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200
  ```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2705 because ADVISOR: Closing — 4L/384d+lr=4e-4 superseded by 4L/320d baseline..

**Comment by tcapelle** [comment]
STUDENT zoro:

## Results

**Note:** This experiment was already running when the advisor closed the PR. Posting results for the record.

### Primary Metrics
| Metric | Baseline (PR #2602, lr=5e-4) | This Run (lr=4e-4) | Change |
|--------|------------------------------|---------------------|--------|
| val_primary/surface_rel_l2_pct | 5.73% | **4.70%** (epoch 259) | **-18.0%** |
| test_primary/surface_rel_l2_pct | — | **5.91%** | — |

### Run Details
- **W&B run:** fjjnqpnj
- **Epochs completed:** 268 (180-min budget)
- **Best val epoch:** 259 (4.70%)
- **Still improving at cutoff:** Yes — best at epoch 259/268

### Key Milestones
```
Epoch  10: 28.02%    Epoch 100:  7.35%
Epoch  20: 19.55%    Epoch 150:  6.59%
Epoch  30: 16.40%    Epoch 200:  6.00%
Epoch  50: 11.50%    Epoch 259:  4.70% ← best
```

### Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 4e-4 --cosine-t-max 30 \
  --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 384 --model-heads 6 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --agent zoro --wandb-name "zoro/drivaerml-4L384d-lr4e4" \
  --wandb-group "drivaerml-4L384d-variants"
```

### What happened
lr=4e-4 at 4L/384d achieves **4.70% val** — beating the lr=5e-4 baseline (5.73%) by 18%. This is the best DrivAerML result so far, getting closer to the external AB-UPT target of 3.71%. The lower LR provides smoother convergence over 268 epochs.
